packages/core/test/bookmark-manager.test.ts

```diff
@@ -30,106 +30,33 @@ describe('BookmarkManager', () => {
-
-      const bookmarks = await manager.getBookmarks('neo4j')
-
-      expect(bookmarks).toEqual([])
-    })
-
-    it('should return bookmarks for the given db', async () => {
-      const manager = bookmarkManager({
-        initialBookmarks: new Map([
-          ['neo4j', neo4jBookmarks],
-          ['system', systemBookmarks]
-        ])
-      })
-
-      const bookmarks = await manager.getBookmarks('neo4j')
-
-      expect(bookmarks).toBeSortedEqual(neo4jBookmarks)
-    })
-
-    it('should return get bookmarks from bookmarkSupplier', async () => {
-      const extraBookmarks = ['neo4j:bm03', 'neo4j:bm04']
-
-      const manager = bookmarkManager({
-        initialBookmarks: new Map([
-          ['neo4j', neo4jBookmarks],
-          ['system', systemBookmarks]
-        ]),
-        bookmarksSupplier: async () => await Promise.resolve(extraBookmarks)
-      })
-
-      const bookmarks = await manager.getBookmarks('neo4j')
-
-      expect(bookmarks).toBeSortedEqual([...neo4jBookmarks, ...extraBookmarks])
-    })
-
-    it('should return not duplicate bookmarks if bookmarkSupplier returns existing bm', async () => {
-      const extraBookmarks = ['neo4j:bm03', 'neo4j:bm04']
-
-      const manager = bookmarkManager({
-        initialBookmarks: new Map([
-          ['neo4j', neo4jBookmarks],
-          ['system', systemBookmarks]
-        ]),
-        bookmarksSupplier: async () => await Promise.resolve([...extraBookmarks, ...neo4jBookmarks])
-      })
-
-      const bookmarks = await manager.getBookmarks('neo4j')
-
-      expect(bookmarks).toBeSortedEqual([...neo4jBookmarks, ...extraBookmarks])
-    })
-
-    it('should return call from bookmarkSupplier with correct database', async () => {
-      const bookmarksSupplier = jest.fn()
-
-      const manager = bookmarkManager({
-        bookmarksSupplier
-      })
-
-      await manager.getBookmarks('neo4j')
-
-      expect(bookmarksSupplier).toBeCalledWith('neo4j')
-    })
-  })
-
-  describe('getAllBookmarks()', () => {
     it('should return all bookmarks ', async () => {
       const manager = bookmarkManager({
-        initialBookmarks: new Map([
-          ['neo4j', neo4jBookmarks],
-          ['system', systemBookmarks]
-        ])
+        initialBookmarks: [...neo4jBookmarks, ...systemBookmarks]
       })
 
-      const bookmarks = await manager.getAllBookmarks()
+      const bookmarks = await manager.getBookmarks()
 
       expect(bookmarks).toBeSortedEqual([...neo4jBookmarks, ...systemBookmarks])
     })
 
-    it('should return empty if there are no bookmarks for any db', async () => {
+    it('should return empty if there are no bookmarks', async () => {
       const manager = bookmarkManager({})
 
-      const bookmarks = await manager.getAllBookmarks()
+      const bookmarks = await manager.getBookmarks()
 
       expect(bookmarks).toBeSortedEqual([])
     })
 
     it('should return enriched bookmarks list with supplied bookmarks', async () => {
```

Comment:
Is there a test to check that the supplied bookmarks don't leak into the state of the BMM?

Reply:
I just added the test